src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Analyzers/ComponentParameterNullableWarningSuppressor.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Razor.Compiler.Analyzers;
+
+#pragma warning disable RS1041 // Compiler extensions should be implemented in assemblies targeting netstandard2.0
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ComponentParameterNullableWarningSuppressor : DiagnosticSuppressor
+{
+    private static readonly LocalizableString Description = new LocalizableResourceString(nameof(AnalyzerResources.ComponentParameterNullableWarningSuppressorDescription), AnalyzerResources.ResourceManager, typeof(AnalyzerResources));
+
+    //Suppress CS8618: "Non-nullable {0} '{1}' must contain a non-null value when exiting constructor. Consider declaring the {0} as nullable."
+    public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions => [
+            new SuppressionDescriptor(AnalyzerIDs.ComponentParameterNullableWarningSuppressionId, "CS8618", Description)
+        ];
+
+    public override void ReportSuppressions(SuppressionAnalysisContext context)
+    {
+        var editorRequiredSymbol = context.Compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Components.EditorRequiredAttribute");
+        var parameterSymbol = context.Compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Components.ParameterAttribute");
+
+        foreach (var diagnostic in context.ReportedDiagnostics)
+        {
+            var node = diagnostic.Location.SourceTree?.GetRoot(context.CancellationToken).FindNode(diagnostic.Location.SourceSpan);
+            if (node is PropertyDeclarationSyntax propertySyntax && propertySyntax.AttributeLists.Any())
+            {
+                var symbol = context.GetSemanticModel(propertySyntax.SyntaxTree).GetDeclaredSymbol(propertySyntax, context.CancellationToken);
+                if (symbol is IPropertySymbol property)
+                {
+                    if (IsEditorRequiredParam(property.GetAttributes()))
```

Comment:
Should we also check that the property has a setter and is `public`? Although for the latter I guess it would be better to have another analyzer saying like "this `[Parameter]` cannot be set at runtime because it's not `public`" (although that might be better as aspnetcore analyzer since it's more tied to the runtime logic).

Reply:
We already have an analyzer in the SDK that errors on those conditions: 
<img width="785" alt="image" src="https://github.com/user-attachments/assets/0aad8846-7355-428a-b55b-69fe9164f25d" />

I'm on the fence as to whether it makes sense to not suppress in those cases. It seems a little weird that you get both a warning and an error, but then you fix the error, and the warning goes away too?

Reply:
Ah, nice. I guess we do not need to suppress in the cases where the analyzer reports an error.